### PR TITLE
expire sessions

### PR DESCRIPTION
**Pull request by grosser**
@jonmoter 

works as expected, tried with `20.seconds` ... did you have any specific default in mind ?

... adding ENV override if anyone requests it, until then a sane default might be good enough ...

### CODE REVIEW

**Comment by sandlerr** [comment]
is this to free up database space? Wouldn't it be better to do that with a cron?

**Comment by grosser** [comment]
Sessions are on Cookies ... This is to limit Session lifetime so users have
to re login once in a while

**Comment by jonmoter** [comment]
If a user is already logged in when this change goes live, what happens?  Will they be logged out, or do you need to create a new session for this change to take effect?

/cc @zendesk/samson

**Comment by grosser** [comment]
if the user has a old session the session will be replaced with one that expires in 1 month

**Comment by grosser** [comment]
hmmm actually if you keep using samson the session will never expire and always be reset to 1 month from last usage ...

**Comment by grosser** [comment]
default rails session expire_after refreshes when used, so not useful to us
this will also add visibility into how samson is used

Initially all users will be logged out, so we get reliable data and do not
keep old users logged in forever.

**Comment by grosser** [comment]
@zendesk/samson FYI this will log out every user once

File: db/schema.rb [hunk 1]
```diff
@@ -477,22 +477,22 @@
...
     t.boolean  "integration",            default: false,      null: false
     t.boolean  "access_request_pending", default: false,      null: false
-    t.string   "time_format",            limit: 255, default: "relative", null: false
+    t.string   "time_format",            default: "relative", null: false
+    t.datetime "last_login_at"
+    t.index ["external_id", "deleted_at"], name: "index_users_on_external_id_and_deleted_at", length: {"external_id"=>191, "deleted_at"=>nil}, using: :btree
   end
 
-  add_index "users", ["external_id", "deleted_at"], name: "index_users_on_external_id_and_deleted_at", length: {"external_id"=>191, "deleted_at"=>nil}, using: :btree
-
```
Comment (jonmoter) [inline db/schema.rb]: Why is this index getting deleted?

File: config/initializers/warden.rb [hunk 2]
```diff
@@ -1,10 +1,11 @@
 # frozen_string_literal: true
+# tested via test/integration/authentication_test.rb
 require 'warden'
 
 Warden::Manager.serialize_into_session(&:id)
 
 Warden::Manager.serialize_from_session do |id|
-  User.find(id)
+  User.where('last_login_at > ?', 1.month.ago).find_by_id(id)
```
Comment (jonmoter) [inline config/initializers/warden.rb]: Let's make this an ENV var that can be configured, with a default to 30 days.

File: config/application.rb [hunk 3]
```diff
@@ -153,8 +153,8 @@ class Application < Rails::Application
 
     config.active_job.queue_adapter = :sucker_punch
     config.samson.export_job = ActiveSupport::OrderedOptions.new
-    config.samson.export_job.downloaded_age = (ENV['EXPORT_JOB_DOWNLOADED_AGE'] || 12.hours).to_i
-    config.samson.export_job.max_age = (ENV['EXPORT_JOB_MAX_AGE'] || 1.day).to_i
+    config.samson.export_job.downloaded_age = Integer(ENV['EXPORT_JOB_DOWNLOADED_AGE'] || 12.hours)
```
Comment (grosser) [inline config/application.rb]: FYI @rikeoka much safer way of loading ... so blank / `1.day` etc fail

**Pull request merged**